app/controllers/tasks_controller.rb

```diff
@@ -138,4 +164,21 @@ def task_params
   def import_confirm_params
     params.permit(:import_id, :subfile_id, :import_type)
   end
+
+  def user_for_api_request
+    authorization_header = request.headers['Authorization']
+    api_key = authorization_header&.split(' ')&.second
+    user_by_api_key(api_key)
+  end
```

Comment:
We recently integrated [Devise](https://github.com/heartcombo/devise) with #582. I haven't checked, but it might provide some handy methods to handle the authorisation via API tokens as well.

Reply:
That's a good idea, but let's take a look at that at a later date and move this to an issue for now